### ISSUE TITLE
Bug: Vue JS Color-Picker not Rendering

### ISSUE THREAD

**Issue by joeyHarig** [Esri/calcite-components#4481]
### Actual Behavior

The calcite-color-picker component doesn not render within a Vue application. When attempting to use the color-picker, the following error occurrs:

<img width="646" alt="Screen Shot 2022-05-02 at 8 48 21 PM" src="https://user-images.githubusercontent.com/55154968/166398521-1b492370-ad64-498d-8a05-f1aefc9a3421.png">

This error is only persistent in the latest version of the software `1.0.0-beta.81`. When reverting to `1.0.0-beta.80`, the color-picker component renders fine.

### Expected Behavior

The calcite-color-picker should render within Vue apps.

### Reproduction Sample

https://github.com/joeyHarig/cc-vue-color-picker-bug

### Reproduction Steps

1. Fork the bug demo repo https://github.com/joeyHarig/cc-vue-color-picker-bug
2. Run `npm install`
3. Run `npm run serve` 
4. Open up the App.vue file to validate that the calcite-date-picker, calcite-color-picker, and calcite-button are included in the markup.
5. View the application on port `http://localhost:8080/`
6. Notice that the color picker component doesn't render.
7. Open the dev tools console to view the error message "TypeError: Cannot use 'in' operator to search for '0' in rgb
    at __spreadArray"

### Reproduction Version

1.0.0-beta.81

#### Info
Browser: Chrome, Firefox
OS: macOS Monterey

#### Resources
Using Stencil within a Vue App: https://stenciljs.com/docs/vue

### Regression?

1.0.0-beta.80

**Comment by macandcheese** [comment]
I'm also seeing this error with CalciteColorPickerHexInput and CalciteColorPickerSwatch locally in React (Gatsby). 

It occurs using when using either the vanilla component or the wrapped React version:

<img width="1063" alt="Screen Shot 2022-05-03 at 8 55 26 AM" src="https://user-images.githubusercontent.com/4733155/166490522-3525b36b-3d04-48b7-80e8-11eb92b63ab1.png">

**Comment by jcfranco** [comment]
### TL;DR - the reported error is caused by a `tslib` bug (used by Stencil build) + `color` lint fix that made it run into the aforementioned bug. See potential fixes near the end.

I looked into this today and here are some findings:

* `color` [fixed some linting errors](https://github.com/Qix-/color/commit/c7b8e759f384748e906943c09255ec8779ebbb6d#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R17) that switched some code to use the array spread operator.
* `tslib` [updated a helper](https://github.com/microsoft/tslib/pull/151) that breaks ☝️ (https://github.com/microsoft/tslib/issues/175).
* Stencil's build uses ☝️ to generate the `esm-es5` output.
* The `esm-es5` code is picked up by Vue's webpack build because of the [default `mainFields`](https://webpack.js.org/configuration/resolve/#resolvemainfields). 

### Solutions 

* As a current workaround for Vue (and possibly other environments), the build config can be updated to specify the `mainFields` to use any of the [following `esm` entry point files](https://github.com/Esri/calcite-components/blob/master/package.json#L7-L9) (`"es2017"`, `"es2015"`, `"jsnext:main"`):

  ```js
  // vue.config.js
  module.exports = {
    chainWebpack: (config) => {
      //...
    
      config.resolve.mainFields
        .merge(["es2017", "browser", "module", "main"]);
    }
  };
  ```

  **Note**: there could be other ways of specifying this, but this is what ended up working for me with the repro case. 
* For the `tslib` bug, I've submitted a PR. If it gets merged, the next step would be to see if we can get the Stencil build to use the latest version. 
* We'd have to look into this beforehand, but we could also update our `module` entry from `package.json` to point to `dist/esm/index.js`. 
* Worst case scenario, and I'd really like to avoid this, we could revert https://github.com/Esri/calcite-components/pull/4381 and https://github.com/Esri/calcite-components/pull/4385 to use a version of `color` that wouldn't run into this issue.

**Comment by jcfranco** [comment]
> We'd have to look into this beforehand, but we could also update our module entry from package.json to point to dist/esm/index.js.

@benelan Let's sync up on this tomorrow to see if we can make the switch.

**Comment by jcfranco** [comment]
Ended up patching `__spreadArray` instead of switching the default `module` entry to minimize impact to builds using this output.
